File: pypann/pypann/Runners/RandomizeWeightsGauss.py
#
# RandomizeWeightsGauss
#

import random
import math
from ..Core.Attributes import Attributes

default_min = -0.3
default_max = +0.3

def randomize_weights_gauss(neuron, net):
    if not hasattr(net, "weight_randomization_attributes"):
        net.weight_randomization_attributes = Attributes()
        net.weight_randomization_attributes.min = default_min
        net.weight_randomization_attributes.max = default_max

    for l in neuron._links_in:
        l.weight()._value = random.uniform(net.weight_randomization_attributes.min, net.weight_randomization_attributes.max)

File: src/pann/Algorithms/RandomizeWeightsGauss.cpp
//RandomizeWeightsGauss.cpp

#include <boost/foreach.hpp>

#include "RandomizeWeightsGauss.h"

REGISTER_SINGLETON_CPP(RandomizeWeightsGaussRunner);

#include "Core/Random.h"
#include <iostream>
namespace pann {
    void
    RandomizeWeightsGaussRunner::run(Object const * net, NeuronPtr neuron) const
    {
        Float min = net->get_attr(hash("RW_MIN"));
        Float max = net->get_attr(hash("RW_MAX"));

        //Tune bias values
        if(neuron->bias)
            neuron->bias->add_value(pann::rand(min, max) - neuron->bias->get_value());

        //Link weights
        BOOST_FOREACH(const Link& link, neuron->input_links)
            link.get_weight()->add_value(pann::rand(min, max) - link.get_weight()->get_value() );
    }; //run

    void
    randomize_weights_gauss(NetPtr net, Float min, Float max)
    {
        net->set_attr(hash("RW_MIN"), min);
        net->set_attr(hash("RW_MAX"), max);
        net->run(RandomizeWeightsGaussRunner::Instance(), Net::ForwardRun);
    }; //randomize_weights_gauss

}; //pann
